in-progress/5040-native-merkle-trees-napi.md

```diff
@@ -0,0 +1,323 @@
+|                      |                                   |
+| -------------------- | --------------------------------- |
+| Issue                | [Native Merkle Trees](https://github.com/AztecProtocol/aztec-packages/issues/5040) |
+| Owners               | @alexghr @PhilWindle |
+| Approvers            | @just-mitch @spalladino |
+| Target Approval Date | 2024-07-05 |
+
+## Executive Summary
+
+This document proposes integrating the [Native Merkle Trees database](https://github.com/AztecProtocol/engineering-designs/blob/f9d1a897303c1481c790cecc4616961e1c183622/in-progress/0003-native-merkle-trees.md) directly into the TypeScript project using a native module written in C++ using [Node-API](https://nodejs.org/docs/latest-v18.x/api/n-api.html) rather than message passing.
+
+## Introduction
+
+The original native Merkle tree spec proposed building a `MerkleTreesDb` native binary in C++. The TypScript code would use message passing over streams to communicate with the database. Messages would be passed over stdin/out which would incur the cost of spawning a process and initializing the C++ code every time an operation is performed. This could be optimized by using long-lived processes but it would mean protecting the interface such that only authorized processes can call the interface.
```

Comment:
I don't think we ever intended to spawn a new process every time. IT was always going to be a long-lived process.

Reply:
I itended to remove this paragraph but missed before pushing the commit :facepalm: